futurex_openedx_extensions/dashboard/details/courses.py

```diff
@@ -0,0 +1,77 @@
+"""Courses details collectors"""
+from __future__ import annotations
+
+from typing import List
+
+from common.djangoapps.student.models import CourseEnrollment
+from django.db.models import Count, OuterRef, Q, Subquery, Sum
+from django.db.models.query import QuerySet
+from eox_nelp.course_experience.models import FeedbackCourse
+from lms.djangoapps.certificates.models import GeneratedCertificate
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from futurex_openedx_extensions.helpers.tenants import get_course_org_filter_list, get_tenant_site
+
+
+def get_courses_queryset(tenant_ids: List, search_text: str = None) -> QuerySet:
+    """
+    Get the courses queryset for the given tenant IDs and search text.
+
+    :param tenant_ids: List of tenant IDs to get the courses for
+    :type tenant_ids: List
+    :param search_text: Search text to filter the courses by
+    :type search_text: str
+    """
+    course_org_filter_list = get_course_org_filter_list(tenant_ids)['course_org_filter_list']
+    tenant_sites = []
+    for tenant_id in tenant_ids:
+        if site := get_tenant_site(tenant_id):
+            tenant_sites.append(site)
+
+    queryset = CourseOverview.objects.filter(
+        org__in=course_org_filter_list
+    )
+    search_text = (search_text or '').strip()
+    if search_text:
+        queryset = queryset.filter(
+            Q(display_name__icontains=search_text) |
+            Q(id__icontains=search_text)
+        )
+    queryset = queryset.annotate(
+        rating_count=Count(Subquery(
+            FeedbackCourse.objects.filter(
+                course_id=OuterRef('id'),
+                rating_content__gt=0,
+            ).values('id')
+        ))
+    ).annotate(
+        rating_total=Sum(
+            FeedbackCourse.objects.filter(
+                course_id=OuterRef('id'),
+                rating_content__gt=0,
+            ).values_list('rating_content', flat=True)
+        )
+    ).annotate(
+        enrolled_count=Count(Subquery(
+            CourseEnrollment.objects.filter(
+                course_id=OuterRef('id'),
+                is_active=True,
+            ).values('id')
+        ))
+    ).annotate(
+        active_count=Count(Subquery(
+            CourseEnrollment.objects.filter(
+                course_id=OuterRef('id'),
+                is_active=True,
+            ).values('id')
+        ))
```

Comment:
@shadinaif this is the same result, but with a different name.

Reply:
it's the same because the data is not available right now. We'll get it from Clickhouse in another PR. Related [issue created](https://github.com/nelc/futurex-openedx-extensions/issues/8)